modin/experimental/core/storage_formats/pandas/native_query_compiler.py

```diff
@@ -0,0 +1,1283 @@
+            )
+        )
+        return _register_default_pandas(
+            _register_expanding(pandas.core.window.expanding.Expanding.cov)
+        )(
+            self,
+            fold_axis,
+            expanding_args,
+            other=other_for_default,
+            pairwise=pairwise,
+            ddof=ddof,
+            numeric_only=numeric_only,
+            squeeze_self=squeeze_self,
+            **kwargs,
+        )
+
+    def expanding_corr(
+        self,
+        fold_axis,
+        expanding_args,
+        squeeze_self,
+        squeeze_other,
+        other=None,
+        pairwise=None,
+        ddof=1,
+        numeric_only=False,
+        **kwargs,
+    ):
+        other_for_default = (
+            other
+            if other is None
+            else (
+                other.to_pandas().squeeze(axis=1)
+                if squeeze_other
+                else other.to_pandas()
+            )
+        )
+        return _register_default_pandas(
+            _register_expanding(pandas.core.window.expanding.Expanding.corr)
+        )(
+            self,
+            fold_axis,
+            expanding_args,
+            other=other_for_default,
+            pairwise=pairwise,
+            ddof=ddof,
+            numeric_only=numeric_only,
+            squeeze_self=squeeze_self,
+            **kwargs,
+        )
+
+    def groupby_size(
+        self,
+        by,
+        axis,
+        groupby_kwargs,
+        agg_args,
+        agg_kwargs,
+        drop=False,
+    ):
+        result = _register_default_pandas(_groupby("size"))(
+            self,
+            by=by,
+            axis=axis,
+            groupby_kwargs=groupby_kwargs,
+            agg_args=agg_args,
+            agg_kwargs=agg_kwargs,
+            drop=drop,
+            method="size",
+        )
+        if not groupby_kwargs.get("as_index", False):
+            # Renaming 'MODIN_UNNAMED_SERIES_LABEL' to a proper name
+
+            result.columns = result.columns[:-1].append(pandas.Index(["size"]))
+        return result
+
+    def get_axis(self, axis):
+        return self._modin_frame.index if axis == 0 else self._modin_frame.columns
+
+    def get_index_name(self, axis=0):
+        return self.get_axis(axis).name
+
+    def get_index_names(self, axis=0):
+        return self.get_axis(axis).names
+
+    def set_index_name(self, name, axis=0):
+        self.get_axis(axis).name = name
+
+    def has_multiindex(self, axis=0):
+        if axis == 0:
+            return isinstance(self._modin_frame.index, pandas.MultiIndex)
+        assert axis == 1
+        return isinstance(self._modin_frame.columns, pandas.MultiIndex)
+
+    def isin(self, values, ignore_indices=False, **kwargs):
+        if isinstance(values, type(self)) and ignore_indices:
+            # Pandas logic is that it ignores indexing if 'values' is a 1D object
+            values = values.to_pandas().squeeze(axis=1)
+        if self._shape_hint == "column":
```

Comment:
Should this check be removed as this class doesn't have _shape_hint?

Reply:
As certain methods like is_in can only be executed on series or dataframe based on the _shape_hint, this class also uses _shape_hint, it will be set in columnarize function which is inherited from BaseQueryCompiler, I have added it to class variables to avoid confusion.